core/src/main/thrift/MLSchema.thrift

```diff
@@ -42,6 +42,18 @@ struct Example {
   2: optional FeatureVector context;
 }
 
+struct DictionaryEntry {
+  1: optional i32 index;
+  2: optional double mean;
+  3: optional double scale; 
+}
+
+struct DictionaryRecord {
+  1: optional map<string, map<string, DictionaryEntry>> dictionary;
+  2: optional i32 entryCount
+}
+
+
```

Comment:
Using a type before it's defined is only supported by recent versions of thrift compiler. As far as I know, if you use thrift version 0.8.0 to compile the original MLSchema.thrift file, an error "Type DictionaryRecord has not been defined" will be reported. See this [link](http://webcache.googleusercontent.com/search?q=cache:2OQpHXHwiQUJ:https://mail-archives.apache.org/mod_mbox/thrift-user/201001.mbox/%253C977a11431001150435m5b2f9536l4f835cc0d40d86fd%40mail.gmail.com%253E+&cd=1&hl=en&ct=clnk&gl=us) for more details.